server/routerlicious/packages/lambdas/src/alfred/index.ts

```diff
@@ -635,15 +638,17 @@ export function configureWebSocketServices(
 						[CommonProperties.clientType]: message.details.details?.type,
 					});
 					connectMetric.success(`Connect document successful`);
-				},
-				(error) => {
+				})
+				.catch((error) => {
 					socket.emit("connect_document_error", error);
 					if (error?.code !== undefined) {
 						connectMetric.setProperty(CommonProperties.errorCode, error.code);
 					}
 					connectMetric.error(`Connect document failed`, error);
-				},
-			);
+				})
+				.finally(() => {
+					connectDocumentComplete = true;
```

Comment:
Is there any difference if we set `connectDocumentComplete` as true in the successful scenario only (`.then(...)`) ? Or do we also want to exercise this new logic even when connect document is false?

Reply:
I see. But isn't it also playing a role in the if condition in line 852? If connectDocumentComplete is false, then we call Lumberjack and also call `await connectDocumentP.catch(() => {});` in line 859. On second thought, I think this is exactly what we want to do. I was probably confused at first, thinking whether we still want to call  `await connectDocumentP.catch(() => {});` in line 859 even when the original connectDocument fails (which would still have `connectDocumentComplete` as true). That is not a problem because that's basically the current flow, I think